src/map/script.c

```diff
@@ -16515,6 +16522,53 @@ static BUILDIN(autoequip)
 	return true;
 }
 
+/**
+ * equipidx(<index>{, <item id>})
+ */
+static BUILDIN(equipidx)
+{
+	int nameid = 0, i = script_getnum(st, 2);
+	struct item_data *item_data;
+	struct map_session_data *sd = script->rid2sd(st);
+
+	if (sd == NULL) {
+		script_pushint(st, 0);
+		return true;
+	}
+
+	if (i < 0 || i >= sd->status.inventorySize) {
+		ShowError("buildin_equipidx: Index (%d) should be from 0-%d.\n", i, sd->status.inventorySize - 1);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	if (i < sd->status.inventorySize) {
+		if (script_hasdata(st, 3))
+			nameid = script_getnum(st, 3);
+		else
+			nameid = sd->status.inventory[i].nameid == nameid;
+
+		if ((item_data = itemdb->exists(nameid)) == NULL) {
+			ShowError("buildin_equipidx: Invalid Item ID (%d).\n", nameid);
+			script_pushint(st, 0);
+			return false;
+		}
+
+		if (sd->status.inventory[i].nameid == nameid && sd->status.inventory[i].equip == 0) {
+			script_pushint(st, pc->equipitem(sd, i, item_data->equip));
+			return true;
+		}
+
+		if (sd->status.inventory[i].nameid != nameid)
+			ShowWarning("buildin_equipidx: Item ID (%d) not found at index (%d) (found Item ID %d).\n", nameid, i, sd->status.inventory[i].nameid);
+		else if (sd->status.inventory[i].equip != 0)
+			ShowWarning("buildin_equipidx: Item ID (%d) at index (%d) cannot be equipped.\n", sd->status.inventory[i].nameid, i);
```

Comment:
I just tested this in-game, I have a priest, trying to equip a knife, it runs this line

IT IS OK to have this check,
but we don't have something like `@inventorylist_canequip` or `*getequipcanequipthis`
so should we suppress this warning ?